### PR TITLE
Reject a request with to_email field empty or missing

### PR DESCRIPTION
**Pull request by zagoli**
fixes reacherhq/check-if-email-exists#1326

### CODE REVIEW

File: backend/tests/check_email.rs [hunk 1]
```diff
@@ -104,3 +104,36 @@ async fn test_reacher_secret_correct_secret() {
 	assert_eq!(resp.status(), StatusCode::OK, "{:?}", resp.body());
 	assert_eq!(resp.body(), FOO_BAR_RESPONSE);
 }
+
+#[tokio::test]
+async fn test_reacher_to_mail_empty() {
+	env::set_var("RCH_HEADER_SECRET", "foobar");
+
```
Comment (amaury1729) [inline backend/tests/check_email.rs:111]:
```suggestion
```

No need to set RCH_HEADER_SECRET for this test

File: backend/tests/check_email.rs [hunk 2]
```diff
@@ -104,3 +104,36 @@ async fn test_reacher_secret_correct_secret() {
...
+	let resp = request()
+		.path("/v0/check_email")
+		.method("POST")
+		.header(REACHER_SECRET_HEADER, "foobar")
```
Comment (amaury1729) [inline backend/tests/check_email.rs:115]:
```suggestion
```

File: backend/tests/check_email.rs [hunk 3]
```diff
@@ -104,3 +104,36 @@ async fn test_reacher_secret_correct_secret() {
...
+		.reply(&create_routes(None))
+		.await;
+
+	assert_eq!(resp.status(), StatusCode::BAD_REQUEST, "{:?}", resp.body());
+	assert_eq!(resp.body(), r#"{"message":"to_mail field is required."}"#);
+}
+
+#[tokio::test]
+async fn test_reacher_to_mail_missing() {
+	env::set_var("RCH_HEADER_SECRET", "foobar");
```
Comment (amaury1729) [inline backend/tests/check_email.rs:126]: Same here

File: backend/tests/check_email.rs [hunk 4]
```diff
@@ -104,3 +104,36 @@ async fn test_reacher_secret_correct_secret() {
...
+
+	let resp = request()
+		.path("/v0/check_email")
+		.method("POST")
+		.header(REACHER_SECRET_HEADER, "foobar")
```
Comment (amaury1729) [inline backend/tests/check_email.rs:131]: And here

File: backend/src/routes/check_email/post.rs [hunk 5]
```diff
@@ -18,14 +18,25 @@
...
 
 /// The main endpoint handler that implements the logic of this route.
 async fn handler(body: CheckEmailInput) -> Result<impl warp::Reply, warp::Rejection> {
-	// Run the future to check an email.
-	Ok(warp::reply::json(&check_email(body).await))
+	// The to_email field must be present
+	if body.to_email.is_empty() {
+		Err(warp::reject::custom(errors::ReacherResponseError{
+			code : http::StatusCode::BAD_REQUEST,
+			message : "to_mail field is required.".to_string()
```
Comment (amaury1729) [inline backend/src/routes/check_email/post.rs:33]:
```suggestion
			message : "to_email field is required.".to_string()
```

**Review by amaury1729 — COMMENTED** [review]
Overall LGTM, could you also run a `cargo fmt`?

File: backend/tests/check_email.rs (hunk 1 above)
Comment (zagoli) [reply backend/tests/check_email.rs:111]: If I omit the header bits the test fails with error "Missing request header"

**Comment by zagoli** [comment]
Hi, I just want to let you know that I don't have merge permissions

**Comment by amaury1729** [comment]
Thanks!

**Pull request merged**